src/Discord/Internal/Types/Events.hs

```diff
@@ -41,11 +41,13 @@ data Event =
   | ThreadCreate               Channel
   -- | thread was updated
   | ThreadUpdate               Channel
+  -- | thread member for the current user was updated
+  | ThreadMemberUpdate         GuildId
```

Comment:
Counter to what I said, I'm pretty sure this event gives a thread member structure with a guild id as an additional field; would it be possible to give the thread member and the guild id? additionally, are you sure that this event parses? I feel like you don't parse out the `guild_id` sufficiently.

Reply:
should i create a ThreadMemberUpdate record that is the thread member with a guild id, also being honest i am not very sure if it parses

Reply:
`gateway - received exception [Error in $: Error in $: parsing Snowflake failed, expected String, but encountered Object] while decoding {"t":"THREAD_MEMBER_UPDATE","s":13,"op":0,"d":{"user_id":"?","muted":false,"mute_config":null,"join_timestamp":"2023-12-08T18:40:00.424615+00:00","id":"?","guild_id":"?","flags":0}}` yeah it doesn't parse currently